src/sagemaker/session.py

```diff
@@ -3308,13 +3309,27 @@ def _create_model_request(
                 "container_defs instead."
             )
             warnings.warn(msg, DeprecationWarning)
+            primary_container = resolve_nested_dict_value_from_config(
+                primary_container,
+                [ENVIRONMENT],
+                MODEL_PRIMARY_CONTAINER_PATH,
+                sagemaker_session=self,
+            )
             container_defs = primary_container
 
         role = self.expand_role(role)
 
+        # If list of container_defs is passed in, update the container defs with values from config
+        # else update container_defs with the primary_container values from config
         if isinstance(container_defs, list):
+            update_list_of_dicts_with_values_from_config(
+                container_defs, MODEL_CONTAINERS_PATH, sagemaker_session=self
+            )
             container_definition = container_defs
         else:
+            container_defs = resolve_nested_dict_value_from_config(
+                container_defs, [ENVIRONMENT], MODEL_PRIMARY_CONTAINER_PATH, sagemaker_session=self
+            )
             container_definition = _expand_container_def(container_defs)
```

Comment:
Pending/TBD: where should configs be plugged in from if both config locations are populated and container_defs is empty?